lib/utils/bitarray.c

```diff
@@ -210,6 +210,112 @@ static void set_region(sys_bitarray_t *bitarray, size_t offset,
 	}
 }
 
+int sys_bitarray_popcount_region(sys_bitarray_t *bitarray, size_t num_bits, size_t offset,
+				 size_t *count)
+{
+	k_spinlock_key_t key;
+	size_t idx;
+	struct bundle_data bd;
+	int ret;
+
+	key = k_spin_lock(&bitarray->lock);
+
+	__ASSERT_NO_MSG(bitarray != NULL);
+	__ASSERT_NO_MSG(bitarray->num_bits > 0);
```

Comment:
There is a logical issue in my mind.
You should check `bitarray` is not empty before `bitarray->lock`.

I know other function is written like you, but I infer it is a mistake.
Maybe we need others to confirm.

There is another reason I think it is a mistake.
Consider follow scenario:
If a thread take a lock and then assert, it will hold lock forever

Reply:
Yes, I just copy/pasted the structure from the existing methods.

I see what you mean and agree that it is wrong in theory, but I am not sure how important it is.
I think the process will quit anyways, either with a SEGFAULT or with an assertion error.
And if the process exits, it doesn't matter that the lock isn't released.

But if the others agree, I can change all locking in this file to be more theoretically correct.